pkg/chunk/chunk_store_utils.go

```diff
@@ -149,24 +201,41 @@ func ProcessCacheResponse(chunks []Chunk, keys []string, bufs [][]byte) (found [
 			missing = append(missing, chunks[i])
 			i++
 		} else if chunkKey > keys[j] {
-			level.Debug(util.Logger).Log("msg", "got chunk from cache we didn't ask for")
+			level.Error(util.Logger).Log("msg", "got chunk from cache we didn't ask for")
```

Comment:
Perhaps Info or Warn? This isn't going to cause a request to fail or anything like that.